aspnetcore/tutorials/first-mvc-app/adding-model.md

```diff
@@ -19,6 +19,298 @@ The model classes you create are known as POCO classes (from **P**lain **O**ld *
+  ```
+
+[!INCLUDE [explains scaffold generated params](~/includes/mvc-intro/model4.md)]
+---
+
+<!-- End of tabs                  -->
+
+If you run the app and click on the **Mvc Movie** link, you get an error similar to the following:
+
+``` error
+An unhandled exception occurred while processing the request.
+
+SqlException: Cannot open database "MvcMovieContext-<GUID removed>" requested by the login. The login failed.
+Login failed for user 'Rick'.
+
+System.Data.SqlClient.SqlInternalConnectionTds..ctor(DbConnectionPoolIdentity identity, SqlConnectionString
+```
+
+You need to create the database, and you use the EF Core [Migrations](xref:data/ef-mvc/migrations) feature to do that. Migrations lets you create a database that matches your data model and update the database schema when your data model changes.
+
+<a name="pmc"></a>
+
+## Initial migration
+
+In this section, the following tasks are completed:
+
+* Add an initial migration.
+* Update the database with the initial migration.
+
+# [Visual Studio](#tab/visual-studio)
+
+1. From the **Tools** menu, select **NuGet Package Manager** > **Package Manager Console** (PMC).
+
+   ![PMC menu](~/tutorials/first-mvc-app/adding-model/_static/pmc.png)
+
+1. In the PMC, enter the following commands:
+
+   ```console
+   Add-Migration InitialCreate
+   Update-Database
+   ```
+
+    `Add-Migration InitialCreate`: Generates an *Migrations/{timestamp}_InitialCreate.cs* migration file. Because this is the first migration, the generated class contains code to create the database schema.
+    `Update-Database`: Updates the database to the last migration. The last migration is the `InitialCreate`  class created with the previous command.
+
+[!INCLUDE [explain warning](~/includes/mvc-intro/model-mig.md)]
+
+[!INCLUDE [ more information on the PMC tools for EF Core](~/includes/ef-pmc.md)]
+
+# [Visual Studio Code / Visual Studio for Mac](#tab/visual-studio-code+visual-studio-mac)
+
+Run the following .NET Core CLI commands:
+
+```console
+dotnet ef migrations add InitialCreate
+dotnet ef database update
+```
+
+The `ef migrations add InitialCreate` command generates an *Migrations/{timestamp}_InitialCreate.cs* migration file. Because this is the first migration, the generated class contains code to create the database schema.
+The `ef database update` command updates the database to the last migration. The last migration is the `InitialCreate`  class created with the previous command.
+
+[!INCLUDE [explain warning](~/includes/mvc-intro/model-mig.md)]
+
+[!INCLUDE [ more information on the CLI tools for EF Core](~/includes/ef-cli.md)]
+
+---
+
+### The InitialCreate class
+
+Examine the *Migrations/{timestamp}_InitialCreate.cs* migration file. The `Up` method creates the Movie table and configures `Id` as the primary key. The `Down` method removes the `Up` migration schema changes.
+
+## Examine the context registered with dependency injection
```

Comment:
This doesn't apply to the SQLite instructions, where that was done manually.  Suggest you integrate this text into a VS tab instead of having it outside of tabs.  Or change directions so for all tabs the context is created manually.